runelite-client/src/main/java/net/runelite/client/ui/overlay/WidgetItemOverlay.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2019, Adam <Adam@sigterm.info>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.util.List;
+import net.runelite.api.widgets.WidgetItem;
+
+public abstract class WidgetItemOverlay extends Overlay
+{
+	private final OverlayRenderer overlayRenderer;
+
+	public WidgetItemOverlay(OverlayRenderer overlayRenderer)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.LOW);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.overlayRenderer = overlayRenderer;
+	}
+
+	public abstract void renderItemOverlay(Graphics2D graphics, int itemId, WidgetItem itemWidget);
+
+	@Override
+	public Dimension render(Graphics2D graphics)
```

Comment:
Why not just do instanceof check in OverlayRenderer? Like the whole point of that class is to manage and draw overlays, and it would make way more sense because then this class would not need to receive OverlayRenderer in constructor and OverlayRenderer will not need to expose itemWidgets